### ISSUE TITLE
Crash report: Error: No device found in bootloader mode


### ISSUE THREAD

**Issue by tessel-crashbot** [tessel/t2-cli#1606]
```
Error: No device found in bootloader mode
    at t2-cli\lib\usb-connection.js:367:19
    at new Promise (<anonymous>)
    at USB.Connection._findBootedDeviceBySerialNumber (t2-cli\lib\usb-connection.js:351:10)
    at Timeout.retry [as _onTimeout] (t2-cli\lib\usb-connection.js:328:12)
    at ontimeout (timers.js:458:11)
    at tryOnTimeout (timers.js:296:5)
    at Timer.listOnTimeout (timers.js:259:5)
```

Full report is at [0x109888b5deaf2d7b](http://crash-reporter.tessel.io/crashes?fingerprint=0x109888b5deaf2d7b)

**Comment by ollietb** [comment]
This is when trying to run `t2 update` on a new board
```
C:\Users\ollie>t2 update
INFO Looking for your Tessel...
INFO Connected to Tessel-02A33B5DA499.
INFO New firmware version found... 0.2.0
INFO Updating Tessel-02A33B5DA499 to latest version (0.2.0)...
INFO Downloading files...
INFO Download complete!
INFO Updating OpenWRT (1/2)
INFO Transferring image of size 16.78 MB. This will take 2-4 minutes...
INFO Transfer complete.
INFO Starting OpenWRT update.
INFO Please do not remove power from Tessel.
INFO This process will take at least 2 minutes...
INFO Configuration is saved during update.
INFO Updating firmware (2/2)
ERR! Detected CLI crash Error: No device found in bootloader mode
ERR!     at C:\Users\ollie\AppData\Roaming\nvm\v9.5.0\node_modules\t2-cli\lib\usb-connection.js:367:19
ERR!     at new Promise (<anonymous>)
ERR!     at USB.Connection._findBootedDeviceBySerialNumber (C:\Users\ollie\AppData\Roaming\nvm\v9.5.0\node_modules\t2-cli\lib\usb-connection.js:351:10)
ERR!     at Timeout.retry [as _onTimeout] (C:\Users\ollie\AppData\Roaming\nvm\v9.5.0\node_modules\t2-cli\lib\usb-connection.js:328:12)
ERR!     at ontimeout (timers.js:458:11)
ERR!     at tryOnTimeout (timers.js:296:5)
ERR!     at Timer.listOnTimeout (timers.js:259:5) Error: No device found in bootloader mode
ERR!     at C:\Users\ollie\AppData\Roaming\nvm\v9.5.0\node_modules\t2-cli\lib\usb-connection.js:367:19
ERR!     at new Promise (<anonymous>)
ERR!     at USB.Connection._findBootedDeviceBySerialNumber (C:\Users\ollie\AppData\Roaming\nvm\v9.5.0\node_modules\t2-cli\lib\usb-connection.js:351:10)
ERR!     at Timeout.retry [as _onTimeout] (C:\Users\ollie\AppData\Roaming\nvm\v9.5.0\node_modules\t2-cli\lib\usb-connection.js:328:12)
ERR!     at ontimeout (timers.js:458:11)
ERR!     at tryOnTimeout (timers.js:296:5)
ERR!     at Timer.listOnTimeout (timers.js:259:5)
INFO Crash Reported: http://crash-reporter.tessel.io/crashes?fingerprint=0x109888b5deaf2d7b
```

**Comment by ollietb** [comment]
@allenjwelch yes. I was originally running in Node bash which was working fine until i tried to connect to a wifi network but it couldn't find any (even though there are about 30 different wifi networks in proximity to my office). Then I tried switching to cmd after reading some advice on other issues and updating the firmware. I've gone through the process of running 
1. `t2 restore` - which sometimes works if it can find the device
2. `t2 list` - which sometimes shows a LIBUSB_TRANSFER_STALL error #1484, sometimes shows nothing and occasionally shows the device
3. `t2 update` - which then shows the `No device found in bootloader mode` error above. The red error light is lit on the board.
4. `t2 list` I get the error below
```
C:\Users\ollie>t2 list
INFO Searching for nearby Tessels...
ERR! Detected CLI crash { Error: LIBUSB_TRANSFER_STALL errno: 4 } Error: LIBUSB_TRANSFER_STALL
INFO Crash Reported: http://crash-reporter.tessel.io/crashes?fingerprint=-0x10b11c85b0316573
WARN No Tessels Found.
Terminate batch job (Y/N)? y
```
I've followed a lot of suggestions in other issues, including trying to put it in bootloader mode by holding down the button as it boots, but the red error light comes on and i cant connect to it. As an example of the inconsistent errors, see this gist https://gist.github.com/ollietb/0a2cfbc741542ab44b5206a1a6d90727

**Comment by allenjwelch** [comment]
@ollietb I had a similar issue. Windows OS and 't2 update' crashed when attempting to update firmware with the bootloader error. Haven't seen the LIBUSB_TRANSFER_STALL error though. And each time I used 't2 restore' I would have to unplug and then replug the tessel from/to the usb drive in order for it to find the tessel again. 

Unfortunately I was never able to get the update command to work with Windows. I did however get around this issue by booting up a linux OS to update the firmware - https://github.com/tessel/t2-cli/issues/1599 

Hope this helps
